source/modules/MEI.header.xml

```diff
@@ -107,6 +107,35 @@
       </attDef>
     </attList>
   </classSpec>
+  <classSpec ident="att.perfRes" module="MEI.header" type="atts">
+    <desc>Attributes that define the characteristics and components of the performance resouce.</desc>
+    <attList>
+      <attDef ident="solo" usage="opt">
+        <desc>Marks this instrument or vocal part as a soloist. Do not use this attribute for a solo
+          instrument which is not accompanied.</desc>
+        <datatype>
+          <rng:ref name="data.BOOLEAN"/>
+        </datatype>
+      </attDef>
+      <attDef ident="adLib" usage="opt">
```

Comment:
I'm sorry but I think I do not understand your question. Instruments can be a solo part or not (true|false) and also optional (ad libitum). So an instrument can be ad libitum or not (true|false).

Reply:
My question is: Why use `adLib` and not `optional` as the attribute name? 

It seems like every time you say what it is, you have to also explain that "ad libitum" means "optional":

"Marks this instrument or vocal part as optional (ad libitum)"
"...and also optional (ad libitum)..."

To me, this means that the word "optional" is actually more understandable in its function than the Latin form.

So I was wondering whether you considered naming it just `@optional` (with the same true or false values), and if so, why `@adLib` was chosen instead.

Reply:
I used adLib because ad libitum is the technical term. In the description I used optional because in my opinion it is not the best way to describe a word using the word itsself.

Reply:
I still think there is probably some problems with using `adLib` in the context of an instrument, since it could also mean 'improvised' or 'free' -- As much as I don't like just using dictionaries to argue the point, here is a screenshot of the dictionary entry for `ad lib`:

<img width="1019" alt="Screenshot 2021-11-11 at 11 45 10" src="https://user-images.githubusercontent.com/163183/141284946-bf371f5d-2ad3-4f85-bcc5-b59cbab5eb97.png">

So the question is whether the ad lib instrument is 'optional' (as in, the piece can be performed without it) or 'improvised' (the instrument is mandatory, but the player can improvise their own line).

Reply:
I got your objection. But in my opinion it is still clear. At this point using `@solo` means that a soloist is needed for the performance and not that the part/voice contains solos. So why should `@adLib` mean here the way of performing and not (as it is with `@solo`) a description of the performance resource? Just a though. I am still in favor of using the technical term.

Reply:
I agree with @riedde

Reply:
I think I was commenting on the fact that 'adlib' means "without previous preparation" instead of "optional". Surely an ensemble wouldn't have an 'adlib' instrument that shows up to play unannounced. :-D